src/pnpq/devices/polarization_controller_thorlabs_mpc320.py

```diff
@@ -213,28 +209,29 @@ def set_params(
         jog_step_2: None | int = None,
         jog_step_3: None | int = None,
     ) -> None:
-        replaced_params: dict[str, int | Quantity] = {}
+        # First load existing params
+        params = self.refresh_params()
+        # Replace params that need to be changed
         if velocity is not None:
-            replaced_params["velocity"] = velocity
+            params["velocity"] = velocity
         if home_position is not None:
-            replaced_params["home_position"] = home_position
+            params["home_position"] = home_position
         if jog_step_1 is not None:
-            replaced_params["jog_step_1"] = jog_step_1
+            params["jog_step_1"] = jog_step_1
         if jog_step_2 is not None:
-            replaced_params["jog_step_2"] = jog_step_2
+            params["jog_step_2"] = jog_step_2
         if jog_step_3 is not None:
-            replaced_params["jog_step_3"] = jog_step_3
-        new_params = dataclasses.replace(self.params, **replaced_params)  # type: ignore
+            params["jog_step_3"] = jog_step_3
+        # Send params to device
         self.connection.send_message_no_reply(
             AptMessage_MGMSG_POL_SET_PARAMS(
                 destination=Address.GENERIC_USB,
                 source=Address.HOST_CONTROLLER,
-                velocity=new_params.velocity,
-                home_position=new_params.home_position.magnitude,
-                jog_step_1=new_params.jog_step_1,
-                jog_step_2=new_params.jog_step_2,
-                jog_step_3=new_params.jog_step_3,
+                velocity=params["velocity"],
+                home_position=params["home_position"].magnitude,
+                jog_step_1=params["jog_step_1"],
+                jog_step_2=params["jog_step_2"],
+                jog_step_3=params["jog_step_3"],
             )
         )
         time.sleep(1)
```

Comment:
We can remove this `sleep` now that we're sleeping on every no-reply command in `connection.py`.